From a0d2b6ccaa99c52088068fdfc1dfebafe3e24832 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Fri, 16 Apr 2021 10:20:32 +0530
Subject: [PATCH] i915: Fix CFI issue in show_dynamic_id function

Kernel panic is seen due to CFI failure in show_dynamic_id function.

Fix the issue by returning ssize_t instead of int value.

Tracked-On: OAM-96773
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 drivers/gpu/drm/i915/i915_perf.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_perf.c b/drivers/gpu/drm/i915/i915_perf.c
index 06de39373378..710553464997 100644
--- a/drivers/gpu/drm/i915/i915_perf.c
+++ b/drivers/gpu/drm/i915/i915_perf.c
@@ -3999,8 +3999,10 @@ static ssize_t show_dynamic_id(struct device *dev,
 {
 	struct i915_oa_config *oa_config =
 		container_of(attr, typeof(*oa_config), sysfs_metric_id);
+	ssize_t ret = 0;
 
-	return sprintf(buf, "%d\n", oa_config->id);
+	ret = sprintf(buf, "%d\n", oa_config->id);
+	return ret;
 }
 
 static int create_dynamic_oa_sysfs_entry(struct i915_perf *perf,
-- 
2.17.1

